### PR TITLE
HPCC-22273 Document CentOS dependency installation

### PR DESCRIPTION
**Pull request by g-pan**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Successful Unit Testing/Build test : 
http://10.224.20.18/view/Docs-gp/job/DocBuild-01-GPRepo/

### CODE REVIEW

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 1]
```diff
@@ -287,6 +287,9 @@
           <para>Optionally you can install the package with rpm (recommended
           using the -Uvh options), however then you would have to negotiate
           installing any additional dependencies.</para>
+
+          <para>One way to install the dependencies on a CentOS system:
+          <programlisting>sudo yum install -y epel-release</programlisting></para>
```
Comment (JamesDeFabia) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:292]: Please add lang="bash" to the program listing

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
one comment inline.

**Review by JamesDeFabia — APPROVED** [review]
Looks good from my POV, but does it need a SME to approve it?

**Comment by g-pan** [comment]
perhaps @Michael-Gardner can review

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 2]
```diff
@@ -287,6 +287,9 @@
           <para>Optionally you can install the package with rpm (recommended
           using the -Uvh options), however then you would have to negotiate
           installing any additional dependencies.</para>
+
+          <para>One way to install the dependencies on a CentOS system:
+          <programlisting lang="bash">sudo yum install -y epel-release</programlisting></para>
```
Comment (Michael-Gardner) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:292]: The epel-release repository should be installed before the hpccsystems platform rpm package. We should introduce this para up around line 283 of the current file. At the end of the first paragraph of the CentOS/Red Hat installation instructions.

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 3]
```diff
@@ -278,14 +278,23 @@
...
-          <para>To install the Platform you should have the appropriate
-          permissions to install packages. If you have sudo rights, then you
-          can install the platform using yum.</para>
+          <para>To install the HPCC Systems Platform you should have the
+          appropriate rights and permissions to install packages on your
+          system. </para>
+
+          <para>If you do have the appropriate rights and permissions, then
+          you can install the platform using yum. If you intend to install
+          using yum you should install the epel-release repository beforehand
```
Comment (JamesDeFabia) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:287]: Do you **install** a repository?

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 4]
```diff
@@ -278,14 +278,23 @@
...
+          to ensure any dependencies are resolved. </para>
+
+          <para>One way to install the repository on a CentOS system:
```
Comment (JamesDeFabia) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:290]: The "one way" aside, should precede the bit about the epel-release repo.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Comments inline

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml (hunk 3 above)
Comment (Michael-Gardner) [reply docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:287]: I've always seen it as 'install' for epel-release. It's a package that adds the repository for you.  It's one of the nice things about centos.  Maybe change 'epel-release repository' to 'epel-release package' ?

Comment (JamesDeFabia) [reply docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:287]: I like Michael's suggestion

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 5]
```diff
@@ -278,14 +278,23 @@
...
+          <programlisting lang="bash">sudo yum install -y epel-release</programlisting>The
+          to install the HPCC Systems platform: </para>
```
[the remark this answers is not in the document]
Comment (JamesDeFabia) [reply docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:292]: @greg, this looks like a very reasonable solution to both comments!

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 6]
```diff
@@ -278,14 +278,23 @@
...
-          can install the platform using yum.</para>
+          <para>To install the HPCC Systems Platform you should have the
+          appropriate rights and permissions to install packages on your
+          system.</para>
+
+          <para>If you do have the appropriate rights and permissions, then
+          you can install the platform using yum. If you intend to install
+          using yum you should install the epel-release repository beforehand
+          to ensure any dependencies are resolved.</para>
+
```
Comment (JamesDeFabia) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:289]: could be tighter by removing lines 281-289 and replacing with this
To install the HPCC Systems Platform you should have the appropriate rights and permissions to install packages on your system. One way to install the platform is using yum.

File: docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml [hunk 7]
```diff
@@ -278,14 +278,23 @@
...
 
-          <programlisting>sudo yum install &lt;hpccsystems platform rpm package&gt;</programlisting>
+          <programlisting>sudo yum install -y epel-release 
+sudo yum install -y &lt;hpccsystems platform rpm package&gt;</programlisting>
 
-          <para>Optionally you can install the package with rpm (recommended
-          using the -Uvh options), however then you would have to negotiate
+          <para>Optionally you can install packages with rpm (recommended
+          using the -Uvh options), however you would then have to negotiate
           installing any additional dependencies.</para>
```
Comment (JamesDeFabia) [inline docs/EN_US/Installing_and_RunningTheHPCCPlatform/Installing_and_RunningTheHPCCPlatform.xml:298]: you can remove "additional"

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
comments inline

**Review by JamesDeFabia — APPROVED** [review]
Looks ok from my POV

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Somehow the lang="bash" attribute was dropped from the <programlisting>

**Review by JamesDeFabia — APPROVED** [review]
Good from my POV

**Review by Michael-Gardner — APPROVED** [review]
Looks really good. Approved on my end.

**Comment by ghalliday** [comment]
@g-pan please squash

**Comment by g-pan** [comment]
@ghalliday should be good to merge now?

**Pull request merged**